core-bundle/src/Resources/public/toggle-fieldset.js

```diff
@@ -0,0 +1,68 @@
+window.addEventListener('DOMContentLoaded', function () {
+    const storeState = function (id, table, state) {
+        fetch(window.location.href, {
+            method: 'POST',
+            headers: {
+                'X-Requested-With': 'XMLHttpRequest'
+            },
+            body: new URLSearchParams({
+                action: 'toggleFieldset',
+                id: id,
+                table: table,
+                state: state,
+                REQUEST_TOKEN: Contao.request_token
+            })
+        });
+    };
+
+    const toggleState = function (el, id, table) {
+        el.blur();
+        Backend.getScrollOffset();
+
+        const fs = el.parentNode;
+
+        if (fs.classList.contains('collapsed')) {
+            fs.classList.remove('collapsed');
+            storeState(id, table, 1);
+        } else {
+            const form = fs.closest('form');
+            const inp = fs.querySelectorAll('[required]');
+            let collapse = true;
+
+            for (let i = 0; i < inp.length; i++) {
+                if (!inp[i].value) {
+                    collapse = false;
+                    break;
+                }
+            }
+
+            if (!collapse) {
+                if (typeof (form.checkValidity) == 'function') form.querySelector('button[type="submit"]').click();
+            } else {
+                fs.classList.add('collapsed');
+                storeState(id, table, 0);
+            }
+        }
+    };
+
+    document.querySelectorAll('legend[data-toggle-fieldset]').forEach(function (el) {
+        const fs = el.parentNode;
+        if (fs.querySelectorAll('label.error, label.mandatory').length) {
+            fs.classList.remove('collapsed');
+        } else if (fs.classList.contains('hide')) {
+            fs.classList.add('collapsed');
+        }
+
+        const { id, table } = JSON.parse(el.getAttribute('data-toggle-fieldset'));
+
+        el.addEventListener('click', function (event) {
+            event.preventDefault();
+            toggleState(el, id, table);
+        })
+    });
+
+    AjaxRequest.toggleFieldset = function (el, id, table) {
+        window.console && console.warn('Using AjaxRequest.toggleFieldset is deprecated and will be removed in Contao 6. Add the data-toggle-fieldset attribute instead.');
+        toggleState(el, id, table)
+    };
```

Comment:
Do we really need a BC layer here?

Reply:
I think so. Anyone who copied stuff from DC_Table will include this line and would otherwise break. Since we did not and cannot deprecate it in Contao 4, and since its so simple, I would keep this atm.

Reply:
I would prefer to remove it and add it again in case it causes problems. I think this is also what have agreed on when it comes to removing BC layers.

Reply:
Sorry but I disagree. The feature is already there, and it does not harm. There was currently no deprecation to use the method, so we must not remove it. As I said, I _know_ this will cause problems because I did write multiple custom DCs 😉 
https://github.com/isotope/core/blob/2.7/system/modules/isotope/drivers/DC_ProductData.php#L516
https://github.com/terminal42/contao-DC_Multilingual/blob/main/src/Driver.php#L250

Reply:
Ok. Please resolve the conflicts then, so I can merge the PR.